iotfunctions/pipeline.py

```diff
@@ -2342,11 +2343,25 @@ def _execute_stage(self, stage, df, start_ts, end_ts, entities, register, to_csv
         self.trace_add(msg, created_by=stage, df=newdf)
         return newdf
 
+    def merge_scoped_df(self, df, newdf, category):
+        if category == 'TRANSFORMER':
+            # TODO Resolve duplicate rows issue
+            cols_to_merge = newdf.columns.difference(df.columns)
```

Comment:
This will allow user to add unwanted columns. We should only take output columns in kpi.

Reply:
@herleraja Still working on this issue

Reply:
@herleraja 
Now using stage object to get columns but for fail safe added 
`cols_to_merge = newdf.columns.difference(df.columns)`